### ISSUE TITLE
Feature request: support writing CSV with GeoJSON geometry

### ISSUE THREAD

**Issue by mentin** [OSGeo/gdal#1076]
org2ogr can read CSV files with GeoJSON-encoded geometry values,
but it does not provide a way to write same CSV file with GeoJSON-encoded geometries,
only has an option (AS_WKT) to write CSV with WKT-encoded geometry values

## Expected behavior and actual behavior.

expected to have option geometry=AS_GEOJSON, similar to geometry=AS_WKT

## Steps to reproduce the problem.

N/A

## Operating system

Any

## GDAL version and provenance

Latest source code, https://github.com/OSGeo/gdal/blob/master/gdal/ogr/ogrsf_frmts/csv/ogr_csv.h 
has AS_WKT option, but does not provide any way to force GEOJSON

**Comment by rouault** [comment]
@jratike80 This is an excellent tip. You could add it in the CSV driver documentation, and I believe that would fix the issue.

**Comment by mentin** [comment]
Fantastic tip, I did not know this is possible. Agree, it would help tremendously if this was mentioned in CSV driver documentation.

**Comment by rouault** [comment]
@mentin Wanting to issue a PR for that ? The file to edit is gdal/ogr/ogrsf_frmts/csv/drv_csv.html

**Comment by juanrmn** [comment]
Hi. I see this was closed long ago, but I think the original feature would be very useful for some use cases.

For example, we have implemented an import pipeline which results in a CSV format output. There are only a few cases we need the geom column to be in GeoJSON format, but with this sqlite option we need some extra info about the input source, like the layer name and the specific name of the geometry column. It would be much easier to have this `geometry=AS_GEOJSON` option so that we don't have to add new steps to the pipeline only to change the geometry output format.

I was wondering if there are more people in similar situations so it's worth it to re-open the original issue.
